tidy3d/components/geometry/base.py

```diff
@@ -2881,5 +2916,24 @@ def _surface_area(self, bounds: Bound) -> float:
         individual_areas = (geometry.surface_area(bounds) for geometry in self.geometries)
         return np.sum(individual_areas)
 
+    @cached_property
+    def _normal_2dmaterial(self) -> Axis:
+        """Get the normal to the given geometry, checking that it is a 2D geometry."""
+
+        normals = {geom._normal_2dmaterial for geom in self.geometries}
+
+        if len(normals) != 1:
+            raise ValidationError(
+                "'Medium2D' requires all geometries in the 'GeometryGroup' to "
+                "share exactly one dimension with zero size."
+            )
+        normal = next(iter(normals))
```

Comment:
what is this line doing? can we make it more straightforward?

Reply:
normals is a `set` of `Axis` corresponding to the axes of all of the 2D geometries? 
so we're first checking that they are all aligned on the same axis and then that all of the bounds are at the same normal position?

so is `next(iter(normals))` just taking the (only) element of the set?